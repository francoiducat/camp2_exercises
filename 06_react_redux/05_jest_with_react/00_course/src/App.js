import React, { Component } from 'react';
import logo from './logo.svg';
import Link from './Link';
import CheckboxWithLabel from './CheckboxWithLabel';
import './App.css';

const hyperlink = "https://www.facebok.com"

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <CheckboxWithLabel labelOn={"labelOn"} labelOff={"labelOff"}/>
        <br/><br/>
        <Link page={ hyperlink }><br /> <br />  { hyperlink }</Link>
      </div>
    );
  }
}

export default App;
