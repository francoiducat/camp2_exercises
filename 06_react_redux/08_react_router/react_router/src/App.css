/* Sticky footer styles
-------------------------------------------------- */
html {
  position: relative;
  min-height: 100%;
}
body {
  margin-bottom: 60px; /* Margin bottom by footer height */
}
.footer {
  position: absolute;
  bottom: 0;
  width: 100%;
  height: 60px; /* Set the fixed height of the footer here */
  line-height: 60px; /* Vertically center the text there */
  background-color: #f5f5f5;
}


/* Custom page CSS
-------------------------------------------------- */
/* Not required for template or sticky footer method. */

.container {
  width: auto;
  max-width: 680px;
  padding: 0 15px;
}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
