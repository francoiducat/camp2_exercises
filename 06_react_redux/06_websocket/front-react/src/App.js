import React, { Component } from 'react';
import './App.css';
import Login from './Login';
import Chat from './Chat';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user: "",
      isLoggedIn: false,
      newMessage:"",
      messages: [{content:"Example Message 1"},
                 {content:"Example Message 2"}
                ]};
  }

  displayChat = () => {
    if(!this.state.isLoggedIn) {
      return(
        <Login login={this.state.user} onSubmit={this.onSubmit} handleOnChange={this.handleOnChange} />
      );
    } else {
      return(
        <Chat
          login={this.state.user}
          handleNewMessage={this.handleNewMessage}
          addMessage={this.addMessage}
          displayMessages={this.displayMessages} />);
    }
  }

  handleNewMessage = event => {
    this.setState({newMessage: event.target.value});
  }

  addMessage = event => {
    event.preventDefault();
    const tableau = this.state.messages.slice();
    tableau.push({content : this.state.newMessage});
    this.setState({messages: tableau});
  }

  displayMessages = () => {
    const displayedMessages = this.state.messages.slice();
    const latestMessage = displayedMessages.reverse();
    return(
      latestMessage.map( message => {
        return(<tr key={message.content}><td className="col">{message.content}</td></tr>);
      }));
  }

  handleOnChange = event => {
    this.setState({user: event.target.value})
  }

  onSubmit = event => {
    event.preventDefault();
    this.setState({isLoggedIn: true})
  }

  displayUsers = () => {

  }

  componentDidMount() {
    // We open a connection from the client, related to the server
    const ws = new WebSocket("ws://localhost:4000");
    const nbUsersElem = document.getElementById("nbUsers");

    // What to do when we receive a message?
    ws.onmessage = (event) => {
      console.log("Message: " + event.data + " clients connected");
      nbUsersElem.innerHTML = event.data;
    };

    ws.onopen = (event) => {
      const user = this.state.user;
      ws.send(user);
    };

    // Alert the server that the client is gone
    window.addEventListener("beforeunload", () => ws.send("CLOSE"));
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title mt-5">SLACKY</h1>
        </header>
        <p>There is <span id="nbUsers">0</span> users connected.</p>

        {this.displayChat()}
      </div>
    );
  }
}

export default App;
