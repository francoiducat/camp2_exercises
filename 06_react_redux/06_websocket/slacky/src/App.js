import React, { Component } from 'react';
import './App.css';
import Login from "./Login";
import Chat from "./Chat";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userName: null,
      messages: [],
    };
    // Attaching the websocket to our App so we can reuse it
    this.websocket = new WebSocket("ws://localhost:8080");
  }

  componentDidMount() {
    // Gets new message from WSS if any
    this.websocket.addEventListener("message", event => {
      const message = JSON.parse(event.data);
      console.log("Message from server : ", message);
      switch (message.type) {
        case "CONNECTION_START":
        default:
          return;
        case "MESSAGES":
          this.setState({ messages: message.data });
          return;
      }
    });
  }

  handleUserName = userName => {
    this.setState({ userName: userName});
    this.websocket.send(
      JSON.stringify({
        type: "LOGIN",
        userName: userName
      })
    );
  };

  sendMessage = newmessage => {
    this.websocket.send(
      JSON.stringify({
        type: "NEW_MESSAGE",
        userName: this.state.userName,
        message: newmessage
      })
    );
  };

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title mt-5">SLACKY</h1>
        </header>

          { this.state.userName ? (
            <Chat
              userName={this.state.userName}
              sendMessage={this.sendMessage}
              messages={this.state.messages}
            />
          ): (
            <Login
              userName={this.state.userName}
              handleUserName={this.handleUserName}
            />
          )}

      </div>
    );
  }
}

export default App;
