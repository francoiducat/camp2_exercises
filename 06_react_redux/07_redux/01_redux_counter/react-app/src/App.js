import React, { Component } from 'react';
import { connect } from "react-redux";
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">REDUX</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}
function mapStateToProps(state) {
  return { todos: state.todos }
}

function mapDispatchToProps(dispatch) {
  return {
    increment: () => dispatch({type: "INCREMENT"})
  }
}

const ConnectedApp = connect(mapStateToProps,mapDispatchToProps)(App);
export default App;
