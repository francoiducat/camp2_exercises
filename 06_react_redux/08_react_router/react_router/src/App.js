import React, { Component } from 'react';
import './App.css';
import BasicExample from './Router';
import Features from './Features';
import { Navbar, Pricing, SubMenuCategories } from './Navbar';
import { Dashboard } from './Dashboard';
import { Footer } from './Footer';

import {
  BrowserRouter as Router,
  Route, NavLink,
  Link, Switch
} from 'react-router-dom'



class App extends Component {
  render() {
    return (
      <div className="Container container-fluid">
      <main className="Container">
        <Navbar />

        <div className="container">
          <Route path="/dashboard" component={Dashboard}/>
          <Route path="/features" component={Features}/>
          <Route path="/pricing" component={Pricing}/>
          <Route path="/categories" component={SubMenuCategories}/>

        </div>

        <hr />

        <p className="App-intro">
          React router basic example :
        </p>
        <BasicExample />




      </main>
      <Footer />
      </div>

    );
  }
}

export default App;
