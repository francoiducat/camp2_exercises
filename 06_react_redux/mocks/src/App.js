import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
const OPENWEATHER_APIKEY = process.env.REACT_APP_OPENWEATHER_APIKEY;
const fetch = require("node-fetch");


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      temp : "Loading...",
      cityName : "Lille"};
  }

  weatherByCityName(city) {
      return fetch(`https://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&APPID=${OPENWEATHER_APIKEY}`)
        .then(response => response.json())
        .then(forecast => forecast.main.temp)
        .catch(error => console.log(error));
  }

  componentDidMount() {
    this.weatherByCityName(this.state.cityName)
          .then(temperature => this.setState({temp : temperature}))
          .catch(e => console.log(e));
    }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Open Weather API Call</h1>
        </header>
          <p>Temperature in { this.state.cityName } is {this.state.temp} Â°C</p>
      </div>
    );
  }
}

export default App;
