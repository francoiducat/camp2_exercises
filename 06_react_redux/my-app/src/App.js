import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import _ from "underscore";


var stooges = [{name: 'moe', age: 40}, {name: 'larry', age: 50}, {name: 'curly', age: 60}];

console.log(_.sortBy(stooges, 'name'));


function displayRow(products) {
  return (
    products.map( product => {
    return (<tr key={ product.decathlon_id}>
      <td>{product.title}</td>
      <td>{product.price}</td>
    </tr>);
  }));
};

const toto = "totototo";
const titi = "titititi"

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      asc : true,
      products: props.products
    }
  }

  sortTable = (filter) => {
    let sortedProducts = _.sortBy(this.state.products, filter)
    if(this.state.asc){
      this.setState({products: sortedProducts, asc: false})
    } else {
      this.setState({products: sortedProducts.reverse(), asc:true})
    }
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Hello Seb !</h1>
        </header>
        <p className="App-intro">
          Hello let's <code>write code/App.js</code> with react js !
        </p>
        <h1>Products</h1>
          <table className="table table-striped">
          <thead>
        <tr>
            <th scope="col" onClick={()=> this.sortTable("title")} >Title</th>
            <th scope="col" onClick={()=> this.sortTable("price")} >Price</th>
        </tr>
      </thead>
        <tbody>

{ displayRow(this.state.products) }

    </tbody>

    </table>
      </div>
    );
  }
}


export default App;
