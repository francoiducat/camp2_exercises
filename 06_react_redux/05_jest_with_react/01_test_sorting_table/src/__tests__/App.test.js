import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import renderer from "react-test-renderer";
import App from '../App';



const products =
[
  { "decathlon_id": 8329121, "title": "COUTEAU A PALOURDES", "price": 4.99 },
  { "decathlon_id": 8247793, "title": "PALMES WADERS", "price": 24.99 }
]

beforeAll(() => {
  configure({ adapter: new Adapter() });
})

test('The table is sorted by price', () => {
  const component = shallow(<App lines={products}/>);

  //prices before sorting
  const oldPrices = component.find("td.price").map(node => node.text());
  const floatOldPrices = oldPrices.map(pri => parseFloat(pri));

  // click on sorting button for prices
  component.find("th.priceHeader").simulate("click")

  // prices after sorting
  const newPrices = component.find("td.price").map(node => node.text());
  const floatNewPrices = newPrices.map(pri => parseFloat(pri))

  // test
  expect(floatNewPrices).toEqual(floatOldPrices.sort(function(a, b) { return a - b; }));
});



it("filters by PRice when third TH is clicked", () => {
  const result = shallow(<App lines={products} />);
  result.find("th").at(2).simulate("click");
  expect(result.state().sortedBy).toBe("price");

});


it("shows a table with header ONE called ID", () => {
  const result = shallow(<App />);
  expect(result.find("th").at(0).text()).toBe("ID");
  expect(result.find("th").at(1).text()).toBe("Title");
  expect(result.find("th").at(2).text()).toBe("Price");
});



SNAPSHOT TESTIN

test("everything looks the same", () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("everything looks the same with props", () => {
  const component = renderer.create(<App lines={ [{"decathlon_id": 8282689, "title": "Corne chasse 14cm", "price": 9.99}] } />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});
