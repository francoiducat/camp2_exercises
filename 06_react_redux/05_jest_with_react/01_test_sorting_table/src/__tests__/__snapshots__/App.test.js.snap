// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`everything looks the same 1`] = `
<table>
  <thead>
    <tr>
      <th
        onClick={[Function]}
      >
        ID
      </th>
      <th
        onClick={[Function]}
      >
        Title
      </th>
      <th
        className="priceHeader"
        onClick={[Function]}
      >
        Price
      </th>
    </tr>
  </thead>
</table>
`;

exports[`everything looks the same with props 1`] = `
<table>
  <thead>
    <tr>
      <th
        onClick={[Function]}
      >
        ID
      </th>
      <th
        onClick={[Function]}
      >
        Title
      </th>
      <th
        className="priceHeader"
        onClick={[Function]}
      >
        Price
      </th>
    </tr>
  </thead>
  <tr>
    <td>
      8282689
    </td>
    <td>
      Corne chasse 14cm
    </td>
    <td
      className="price"
    >
      9.99
    </td>
  </tr>
</table>
`;
